code/extensions/che-commands/src/taskProvider.ts

```diff
@@ -37,6 +37,7 @@ export class DevfileTaskProvider implements vscode.TaskProvider {
 
 		const cheTasks: vscode.Task[] = devfileCommands!
 			.filter(command => command.exec?.commandLine)
+			.filter(command => !command.attributes || (command.attributes as any)['controller.devfile.io/imported-by'] !== 'editor')
```

Comment:
I'm worried that this is a somewhat fragile approach; the attribute `controller.devfile.io/imported-by: editor` comes from the name of the plugin/contribution that imported it, so this works only as long as devfiles specify the name `editor` for the plugin. Currently, as far as I can tell, this is done in the devfile registry, [here](https://github.com/eclipse-che/che-devfile-registry/blob/25f630b689d3fbc90e23a798a15e2ef3c9a17002/tools/devworkspace-generator/src/generate.ts#L77-L82), which results in devworkspaces that contain
```yaml
  contributions:
    - name: editor # <--- source for the attribute
      kubernetes:
        name: <workspace-specific-name>
```

This may also pose a problem in DevWorkspaces created in an earlier version of Che, as previously the editor contribution was named differently (e.g. in my sandbox workspaces, I have 
```yaml
    commands:
    - apply:
        component: che-code-injector
      attributes:
        controller.devfile.io/imported-by: che-code-golang-example
      id: init-container-command
```


Another approach, that would be safer but more complicated, would be to filter commands that are tied to event bindings in the flattened DevWorkspace. In other words, we could look at
* `.events.prestart[]`
* `.events.poststart[]`
* `.events.prestop[]`

and filter commands if their id appears in any of those lists. Going off the above example, my (flattened) DevWorkspace has the following events:
```yaml
    events:
      postStart:
      - init-che-code-command
      preStart:
      - init-container-command
```